### ISSUE TITLE
Messages with severity="DEBUG" are not output through the Web API

### ISSUE THREAD

**Issue by josteinaj** [daisy/pipeline-modules-common#88]
The [code for `px:message`](https://github.com/daisy/pipeline-modules-common/blob/master/common-utils/src/main/java/org/daisy/common/xproc/calabash/steps/Message.java) is essentially this:

```java
public void run() throws SaxonApiException {
	String severity = getOption(_severity, "INFO");
	String message = getOption(_message, (String)null);
	if ("WARN".equals(severity)) {
		warning(this.step.getNode(), message);
	} else if ("DEBUG".equals(severity)) {
		// DefaultStep#fine() removed in ndw/xmlcalabash1@ce9b07d
		this.runtime.getMessageListener().fine(this,this.step.getNode(), message);
	} else {
		info(this.step.getNode(), message);
	}
```

The DEBUG messages are for some reason not included in the job XMLs `<message>` section.

@capitancambio maybe you have an idea why this is? They appear in the detailed log.

**Comment by bertfrees** [comment]
It's because of https://github.com/daisy/pipeline-framework/blob/master/calabash-adapter/src/main/java/org/daisy/common/xproc/calabash/impl/EventBusMessageListener.java#L89. I asked Javi too but I think he didn't remember why he commented out that line.

**Comment by bertfrees** [comment]
See https://github.com/daisy/pipeline-framework/pull/96

**Comment by josteinaj** [comment]
Was this ever fixed? I don't see any DEBUG messages in my log at the moment but maybe there just aren't any...

**Comment by capitancambio** [comment]
You need to set the property as in :

https://github.com/daisy/pipeline-framework/blob/master/calabash-adapter/src/main/java/org/daisy/common/xproc/calabash/impl/EventBusMessageListener.java#L24

**Comment by josteinaj** [comment]
Ah, ok. That should be set to `true` by default in the assembly then. I assume there won't be many cases where we want to hide the debug messages as they aren't always used as actual debug messages but rather "less significant" messages.

**Comment by capitancambio** [comment]
Dunno,  that might add quite an overhead (in terms of data etc...) for clients, is the progress thingy being sent as debug or info?

**Comment by bertfrees** [comment]
You could set it in https://github.com/snaekobbi/system/blob/master/roles/test-server/templates/opt/daisy-pipeline2/etc/system.properties.

**Comment by bertfrees** [comment]
Also, we should carefully consider whether debug messages are even supposed to end up in the job messages. The progress thingy should be based on INFO messages I think.

**Comment by josteinaj** [comment]
@bertfrees ok, thanks, let's do that then.

@capitancambio Progress messages can be sent with any severity, but should be sent with a debug severity to avoid cluttering the UI.

I think we need a way to distinguish messages that say "now I'm starting some main script component" and "now I'm doing some minor stuff", to avoid spamming the user with too much messages. If we want this distinction and still remove debug messages then we either need to add another severity level, or maybe we could use the progress mechanism to distinguish between important and non-important messages since the progress mechanism can be used to determine the "depth" of a message.

**Comment by capitancambio** [comment]
> we could use the progress mechanism to distinguish between important and non-important messages since the progress mechanism can be used to determine the "depth" of a message.

I reckon we should use always info for progress messages but filter them out in the clients, it's less overhead than having the debug messages, what you think?

**Comment by josteinaj** [comment]
@capitancambio Yes, the progress part of the messages are already filtered out, and I think if there are no message after the progress part of the message then the message is not passed on to the webui. But do you think we don't need a separation between "important" and "not important" messages?

If we want to use the progress mechanism to determine the "depth" of a message it would be so that this series of messages:

```
[progress 10 px:dtbook.load] Loading DTBook fileset
[progress dtbook.load 50] Loading DTBook XML
[progress dtbook.load 50] Loading DTBook resource files
[progress 70 px:dtbook-to-pef] Converting DTBook to PEF
Message with no progress info
[progress px:dtbook-to-pef 10 px:dtbook-validate] Validating input DTBook
[progress px:dtbook-validate 50 px:xml-validate-well-formed] Checking well-formedness of DTBook
Another message with no progress info
[progress px:xml-validate-well-formed 100] Checking well-formedness of XML
```

Would translate to this message hierarchy:

- Loading DTBook fileset
  - Loading DTBook XML
  - Loading DTBook resource files
- Converting DTBook to PEF
  - Message with no progress info
  - Validating input DTBook
    - Checking well-formedness of DTBook
      - Another message with no progress info
      - Checking well-formedness of XML

Where we could say that the important messages are:

- Loading DTBook fileset
- Converting DTBook to PEF

(and maybe one level deeper as well)

**Comment by bertfrees** [comment]
Also, in the light of unifying the two kinds of messages and dispatching based on log level, like I just proposed in https://github.com/daisy/pipeline-issues/issues/478, it would be good to not mix debug messages that you want in the execution log with debug messages that you don't want in the execution log (there are a lot!).

Maybe indeed if you want a hierarchy we can make use of the progress mechanism, instead of log levels. I like that. You can have an infinite number of levels, while there's only one INFO.

**Comment by josteinaj** [comment]
I'll add a `depth` attribute to messages in the clientlib-java execution logs in case we want to use the messaging approach.

I've changed almost all of our messages to have DEBUG severity in the v1.10 PRs. Should I change them back to INFO?

**Comment by bertfrees** [comment]
Aha, right, and you did that because we decided in Nancy we wanted some kind of hierarchy of log messages but we didn't know of a better way to accomplish that at the time, right?  I would say yes, revert that, but maybe let's take some time to think about it before we make premature decisions again :)

**Comment by josteinaj** [comment]
It might be that we decided in Nancy, I don't remember but it sounds likely. I'll wait with reverting to INFO for all the messages for now.

**Comment by josteinaj** [comment]
I just finished writing up an initial version of the progress message documentation here:

https://github.com/daisy/pipeline-clientlib-java/wiki/Progress-statements
